core/src/dird/autoprune.cc

```diff
@@ -90,7 +88,7 @@ void PruneVolumes(JobControlRecord* jcr,
   int count;
   UaContext* ua;
   dbid_list ids;
-  del_ctx prune_list;
+  std::vector<JobId_t> prune_list;
```

Comment:
```suggestion
  std::vector<JobId_t> prune_list(10'000);
```

Reply:
We could also preallocate 10'000 JobIds in the prune_list as it was in the old code.

Reply:
the old code set the 10000 as a maximum value. If we preallocate with `prune_list(10'000)`, the prune_list will be preset  with zeros, and then values will be pushed back beyond that value. It doesn't seem logic, unless I'm missing something?

Reply:
As far as the internet goes, people seem to converge that letting the vector grow on it's own is the best solution most of the time, unless you are really sure you have huge quantities to store.
So, unless we are sure that we prune huge quantities of jobs every time (and if I'm not wrong, 10'000 is not a big deal in 2022), it doesn't make sense to preallocate things.

Reply:
Agreed